test/cpu_reducecalculation.hpp

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/miopen.h>
+#include <miopen/reducecalculation.hpp>
+
+#include "tensor_holder.hpp"
+
+#include "../src/kernels/MIOpenReduceCalculation.hpp"
+
+template <typename T, ReduceCalculationOp_t op>
+void cpu_calculation_forward(tensor<T> input,
+                             tensor<T>& ref_output,
+                             int32_t dim,
+                             miopenReduceCalculationNanPropagation_t nanPropagation)
```

Comment:
```suggestion
void cpu_calculation_forward(const tensor<T>& input,
                             tensor<T>& ref_output,
                             int32_t dim,
                             miopenReduceCalculationNanPropagation_t nanPropagation)
```

Reply:
What's the reason to pass input as reference? I only use its values, I dont modify or return input later

Reply:
I only agree with using `const` for input, but not `const&`

Reply:
Since this is just a test file, I don’t think using references for performance improvement is necessary. Is there any MIOpen convention that specifies this? If not, let’s keep it as-is for simplicity